controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +134,335 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+			if existing.DeletionTimestamp != nil {
+				return fmt.Errorf("failed to restore StorageClass  %s because it is marked for deletion", existing.Name)
+			}
+			if !reflect.DeepEqual(storageClass.Parameters, existing.Parameters) {
+
+				// Since we have to update the existing StorageClass, so we will delete the existing storageclass and create a new one.
+				r.log.Info("StorageClass needs to be updated, deleting it.", "StorageClass", klog.KRef(storageClass.Namespace, existing.Name))
+				err = r.Client.Delete(r.ctx, existing)
+				if err != nil {
+					r.log.Error(err, "Failed to delete StorageClass.", "StorageClass", klog.KRef(storageClass.Namespace, existing.Name))
+					return err
+				}
+				r.log.Info("Creating StorageClass.", "StorageClass", klog.KRef(storageClass.Namespace, storageClass.Name))
+				err = r.Client.Create(r.ctx, storageClass)
+				if err != nil {
+					r.log.Info("Failed to create StorageClass.", "StorageClass", klog.KRef(storageClass.Namespace, storageClass.Name))
+					return err
+				}
+			}
+		}
+	}
+	if len(skippedStorageClass) > 0 {
+		return fmt.Errorf("some StorageClasses [%s] were skipped while waiting for pre-requisites to be met", strings.Join(skippedStorageClass, ","))
+	}
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) newCephBlockPoolStorageClassConfiguration(storageClassConfiguration StorageClassConfiguration) StorageClassConfiguration {
+	persistentVolumeReclaimDelete := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	managementSpec := r.storageCluster.Spec.ManagedResources.CephBlockPools
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-ceph-rbd", r.storageCluster.Name),
+			Annotations: map[string]string{
+				"description": "Provides RWO Filesystem volumes, and RWO and RWX Block volumes",
+			},
+		},
+		Provisioner:   fmt.Sprintf("%s.rbd.csi.ceph.com", r.storageCluster.Namespace),
+		ReclaimPolicy: &persistentVolumeReclaimDelete,
+
+		// AllowVolumeExpansion is set to true to enable expansion of OCS backed Volumes.
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Parameters: map[string]string{
+			"clusterID":                 r.storageCluster.Namespace,
+			"pool":                      fmt.Sprintf("%s-cephblockpool", fmt.Sprintf("%s-cephblockpool", r.storageCluster.Name)),
+			"imageFeatures":             "layering",
+			"csi.storage.k8s.io/fstype": "ext4",
+			"imageFormat":               "2",
+			"csi.storage.k8s.io/provisioner-secret-name":            "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/provisioner-secret-namespace":       r.storageCluster.Namespace,
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-rbd-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        r.storageCluster.Namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": r.storageCluster.Namespace,
+		},
```

Comment:
These are all defaults, please integrate the custom params we have for the external SC here (passed as an argument)